Panic

```bash
panic: reflect.Value.Interface: cannot return value obtained from unexported field or method [recovered]
	panic: reflect.Value.Interface: cannot return value obtained from unexported field or method
```

```go
package transformer

import (
	"github.com/davecgh/go-spew/spew"
	"github.com/finsmart-gmbh/api/model"
	"github.com/finsmart-gmbh/pointer"
	"github.com/makeless/makeless-go/model"
	"github.com/r3labs/diff/v2"
	"github.com/stretchr/testify/assert"
	"sync"
	"testing"
)

// dst: database company lead
// src: company lead from company lead transformer
func TestCompanyLeadMergeTransformerTransform(t *testing.T) {
	tests := []struct {
		dst *model.CompanyLead
		src *model.CompanyLead
	}{
		{
			dst: &model.CompanyLead{
				Model: makeless_go_model.Model{Id: 1},
				CompanyLeadHouseholds: []*model.CompanyLeadHousehold{
					{
						Model: makeless_go_model.Model{Id: 2},
						CompanyLeadPersons: []*model.CompanyLeadPerson{
							{
								Model:     makeless_go_model.Model{Id: 3},
								FirstName: pointer.StringPtr("Max"),
								RWMutex:   new(sync.RWMutex),
							},
						},
						RWMutex: new(sync.RWMutex),
					},
				},
				RWMutex: new(sync.RWMutex),
			},
			src: &model.CompanyLead{
				Model: makeless_go_model.Model{Id: 1},
				CompanyLeadHouseholds: []*model.CompanyLeadHousehold{
					{
						Model: makeless_go_model.Model{Id: 2},
						CompanyLeadPersons: []*model.CompanyLeadPerson{
							{
								Model:     makeless_go_model.Model{Id: 3},
								FirstName: pointer.StringPtr("Max"),
								Address: &model.Address{
									Street:  pointer.StringPtr("maxstraße"),
									RWMutex: new(sync.RWMutex),
								},
								RWMutex: new(sync.RWMutex),
							},
							{
								Model:     makeless_go_model.Model{Id: 4},
								FirstName: pointer.StringPtr("Fisch"),
								Address: &model.Address{
									Street:  pointer.StringPtr("maxstraße"),
									RWMutex: new(sync.RWMutex),
								},
								RWMutex: new(sync.RWMutex),
							},
						},
						RWMutex: new(sync.RWMutex),
					},
				},
				RWMutex: new(sync.RWMutex),
			},
		},
	}

	for i, test := range tests {
		changelog, err := diff.Diff(test.dst, test.src)
	}
}
```

--- a/diff_struct.go
+++ b/diff_struct.go
@@ -54,6 +54,11 @@ func (d *Differ) diffStruct(path []string, a, b reflect.Value) error {
 			continue
 		}
 
+		// skip private fields
+		if !a.CanInterface() {
+			continue
+		}
+
 		err := d.diff(fpath, af, bf, a.Interface())
 		if err != nil {
 			return err

--- a/diff_test.go
+++ b/diff_test.go
@@ -6,6 +6,7 @@ package diff
 
 import (
 	"reflect"
+	"sync"
 	"testing"
 	"time"
 
@@ -49,6 +50,11 @@ type embedstruct struct {
 type customTagStruct struct {
 	Foo string `json:"foo"`
 	Bar int    `json:"bar"`
+}
+
+type privateValueStruct struct {
+	Public  string
+	Private *sync.RWMutex
 }
 
 type CustomStringType string
@@ -349,6 +355,13 @@ func TestDiff(t *testing.T) {
 			Changelog{
 				Change{Type: UPDATE, Path: []string{"unidentifiables", "0", "value"}, From: 1, To: 5},
 				Change{Type: CREATE, Path: []string{"unidentifiables", "3", "value"}, From: nil, To: 4},
+			},
+			nil,
+		},
+		{
+			"struct-with-private-value", privateValueStruct{Public: "one", Private: new(sync.RWMutex)}, privateValueStruct{Public: "two", Private: new(sync.RWMutex)},
+			Changelog{
+				Change{Type: UPDATE, Path: []string{"Public"}, From: "one", To: "two"},
 			},
 			nil,
 		},

